rhino/src/main/java/org/mozilla/javascript/ThreadSafeCompoundOperationMap.java

```diff
@@ -0,0 +1,101 @@
+package org.mozilla.javascript;
+
+import java.util.Iterator;
+
+class ThreadSafeCompoundOperationMap extends CompoundOperationMap {
```

Comment:
A little bit of Javadoc to explain what this class is for and how it relates to other classes would be super helpful